src/Motor.Extensions.Diagnostics.Metrics/AutoIncCounter.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Prometheus.Client;
+
+namespace Motor.Extensions.Diagnostics.Metrics
+{
+    public class AutoIncCounter : IDisposable
+    {
+        private readonly Func<ICounter?> _counterDelegate;
+
+        public AutoIncCounter(Func<ICounter?> counterDelegate)
```

Comment:
nit: a doc comment about the usage of the delegate could be helpful to at least explain the reason for the whole thing regarding the "fix" Rider offers for this but just an idea

Reply:
I will extend the existing comment in the usage example